client/app/lib/nodenotificationclient.coffee

```diff
@@ -0,0 +1,136 @@
+debug = require('debug') 'app:nodenotification'
+
+kd = require 'kd'
+
+SockJS = require 'sockjs-client'
+backoff = require 'backoff'
+
+kookies = require 'kookies'
+globals = require 'globals'
+
+nick = require 'app/util/nick'
+
+
+module.exports = class NodeNotificationClient extends kd.Object
+
+  SUBSCRIBE_ENDPOINT = '/notify/subscribe' # this can be taken from config ~ GG
+
+
+  constructor: (options = {}, data) ->
+
+    super options, data
+
+
+    @_bo = backoff.exponential
+      initialDelay : 1200
+      maxDelay     : 15000
+
+    @on 'ready', =>
+      debug 'connected and verified, listening for new events...'
+      @_bo.reset()
+
+    @_bo.failAfter 15
+
+    @_bo.on 'backoff', (trial, delay) ->
+      return  if trial is 0
+      debug "failed to connect on trial #{trial}, trying again in #{delay}ms..."
+
+    @_bo.on 'fail', ->
+      debug 'failed to connect, giving up.'
+
+
+  connect: ->
+
+    @_bo.on 'ready', =>
+      @_socket?.close()
+      delete @_socket
+      @_connect()
+
+    @_bo.backoff()
+
+
+  _connect: ->
+
+    debug 'initiating the socket connection...'
+    @_socket = new SockJS SUBSCRIBE_ENDPOINT
+
+    @_socket.onmessage = @bound '_handleNewMessage'
+    @_socket.onclose   = @bound '_handleClose'
+    @_socket.onopen    = @bound '_handleOpen'
```

Comment:
errors are not handled if they don't end with close so I ended up with using close only, what should we do when error happened but connection is still open?

Didn't your last comment though :) how it will help to address the issue?